app/controllers/redirections_controller.rb

```diff
@@ -19,7 +19,13 @@ def previous
   def find_or_create_redirection
     Redirection.find_by(slug: params[:slug]) ||
       RedirectionCreation.perform(referrer, params[:slug]) ||
-      Redirection.first
+      log_headers_and_use_fallback_redirection
+  end
+
+  def log_headers_and_use_fallback_redirection
+    logger.info "Redirection creation failed. Here are the headers:"
+    logger.info request.env
```

Comment:
This has a lot of stuff that isn't just `headers` - can we trim it down to just the headers?

Reply:
Screenshot of everything that gets logged:

<img width="1420" alt="hotline-webring-production - Papertrail 2020-09-26 12-06-42" src="https://user-images.githubusercontent.com/257678/94348401-c88ae080-fff0-11ea-913a-3cbf9294c0d2.png">

Reply:
One option, though not totally foolproof, is to only select keys that are all uppercase:

```ruby
headers = request.env.select { |(key, _)| key.upcase == key }
```

Reply:
Ah, there is a foolproof method, from [the Rack spec](https://github.com/rack/rack/blob/master/SPEC.rdoc):

>`HTTP_` Variables
>Variables corresponding to the client-supplied HTTP request headers (i.e., variables whose names begin with `HTTP_`). The presence or absence of these variables should correspond with the presence or absence of the appropriate HTTP header in the request. See RFC3875 section 4.1.18 for specific behavior.

So we can do this:


```ruby
headers = request.env.select { |(key, _)| key.start_with?("HTTP_") }
```

Reply:
And to cut down on noise, we might want to remove `HTTP_COOKIE`, which is a long string like this:

```
"HTTP_COOKIE"=>"_hotline-webring_session=d3M5MzQzdVhHeW5uSnVaWmNNdXpLYUg5a0NnWGFJWWdKckRNWXlGN2hTbTB5TTBIVnExcUhHcGJvaU03ZHBySitndnY0V3BlWnYrSTJlTGNxU3g3WUt1MEgvcWVoVWl0aGt5NGRqQmN4UVJMYWZGZGNqSXVXUEx4UXh0UUpyT0lWVFpqbDF2OThmWnhvQldQMnZpeWJ3PT0tLWwwMVZMeCtyeTk5WVJXNjk4NWV6aVE9PQ%3D%3D--1851b3c61a861f212c969ddaa493b5dd21471abb"
```

```ruby
headers = request.env.select { |(key, _)| key.start_with?("HTTP_") && key != "HTTP_COOKIE" }
```

Reply:
I also found https://stackoverflow.com/a/51884276/638966 which mentions that effectively no HTTP header has a `.` in it. This does show us a few other headers but nothing that I think we'll care about or will be different from request to request. I like your HTTP but not COOKIE idea.

Reply:
Boom dc93b2b